Sources/MessagingInApp/Gist/Managers/MessageQueueManager.swift

```diff
@@ -1,13 +1,34 @@
+import CioInternalCommon
 import Foundation
 import UIKit
 
-class MessageQueueManager {
+protocol MessageQueueManager: AutoMockable {
+    var interval: Double { get set }
+    func setup()
+    // sourcery:Name=setupSkipQueueCheck
+    // sourcery:DuplicateMethod=setup
+    func setup(skipQueueCheck: Bool)
+    func fetchUserMessagesFromLocalStore()
+    func removeMessageFromLocalStore(message: Message)
+    func clearUserMessagesFromLocalStore()
+    func getInlineMessages(forElementId elementId: String) -> [Message]
+}
+
+// sourcery: InjectRegisterShared = "MessageQueueManager"
+class MessageQueueManagerImpl: MessageQueueManager {
     var interval: Double = 600
     private var queueTimer: Timer?
-    // The local message store is used to keep messages that can't be displayed because the route rule doesnt match.
-    private var localMessageStore: [String: Message] = [:]
+    // The local message store is used to keep messages that can't be displayed because the route rule doesnt match and inline messages.
+    var localMessageStore: [String: Message] = [:]
```

Comment:
public mutable property is almost always not good choice.
let alone the no-threading-protection

Reply:
The property is `internal` as that's the swift default. It's made internal so we can reference it in our automated tests. 

As far as no-threading, that's a good call-out. I can make it have safe reading and writing access as we do in the codebase today.

Reply:
I [updated](https://github.com/customerio/customerio-ios/pull/716/files#diff-ec01291f32e74af14510929ca3d297f876bb1882bd4fdd5008cadca4dff331bcR19-R23) the `localMessageStore` and `interval` to have atomic read and write access.